spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -457,6 +463,18 @@ public void setIdleReplyContainerTimeout(long idleReplyContainerTimeout, TimeUni
 		this.idleReplyContainerTimeout = unit.toMillis(idleReplyContainerTimeout);
 	}
 
+	/**
+	 * Set to true to receive replies asynchronously on the reply container thread.
+	 * Requires {@link #setUseReplyContainer(boolean) userReplyContainer} to be true
+	 * and a {@link #setCorrelationKey(String) correlationKey} (usually 'JMSCorrelationID').
+	 * This is ignored otherwise.
+	 * @param async the async to set.
+	 * @since 4.3
+	 */
+	public void setAsync(boolean async) {
```

Comment:
I think after merging #1739 we can remove this setter and its property altogether.
Having the `protected getAsyncReplySupported()` we good to go with your internal logic around `this.correlationKey`.